core/ExceptionHandler.php

```diff
@@ -113,11 +113,13 @@ public static function replaceSensitiveValues(string $message): string
         $dbConfig = Db::getDatabaseConfig();
 
         $valuesToReplace = [
-            Piwik::getCurrentUserTokenAuth() => 'tokenauth',
-            SettingsPiwik::getSalt()         => 'generalSalt',
-            $dbConfig['username']            => 'dbuser',
-            $dbConfig['password']            => 'dbpass',
+            Piwik::getCurrentUserTokenAuth() ?: '' => 'tokenauth',
```

Comment:
If we were changing this to allow for empty values, it might be best to switch the keys and values around as we can have multiple empty values but not keys, and we always have the replacement value. Having an empty string as an array key might be valid but not super predictable and could cause some debugging headaches later on.
We could then use array_filter to remove empty values before the actual replacement.

Reply:
Thanks for the input @michalkleiner . In another circumstance, I would likely agree with you. However, I don't see any use case where an empty string would be a valid search value for replacement, especially when trying to mask sensitive data. An empty string isn't sensitive data and wouldn't `str_replace` simply ignore an empty string as the search value?

Reply:
@snake14 as mentioned in my comment, we'd first run the list of replacements via array_filter and it would remove it, so it wouldn't be used to search for a value to be replaced. That mechanism would also cover any other replacements that don't actually provide a value, so would be a bit more robust mechanism.